agent/node.go

```diff
@@ -74,6 +74,11 @@ type NodeConfig struct {
 	// HeartbeatTick defines the amount of ticks between each
 	// heartbeat sent to other members for health-check purposes
 	HeartbeatTick uint32
+
+	// Spec is the spec of the node being requested, not all fields in
```

Comment:
Discussed this earlier and we think using a `Spec` is better than putting specific fields like `Availability` into the `IssueNodeCertificateRequest`. The spec is the structure that contains all these fields, so we shouldn't duplicate it by building a parallel structure inside `IssueNodeCertificateRequest`.